# SAAS---Sex-associated Alternative Splicing
## Preparing
**you can process your data with this script, here we provide a demo for users**<br>
**Notes: you can download this script and run it in your computer(need to download gtf/gff files. [Mus_musculus](http://ftp.ensembl.org/pub/release-97/gtf/mus_musculus/Mus_musculus.GRCm38.97.gtf.gz))**
1. [python 3.7](https://www.python.org/)
2. [R 3.6.0](https://cran.r-project.org/mirrors.html)
3. you got your RNA-seq analysis data.--- [example](https://github.com/leequn/saas/tree/master/exp_kidney)
4. you got your alternative splicing analysis data by [rMATS](http://rnaseq-mats.sourceforge.net/).--- [example](https://github.com/leequn/saas/tree/master/as_kidney)
5. sex samples lable. ---[example: female](https://github.com/leequn/saas/blob/master/kidney_female.txt), [example: male](https://github.com/leequn/saas/blob/master/kidney_male.txt)
6. sample information(i.e. development age, tissues).---[example](https://github.com/leequn/saas/blob/master/info_kidney.txt)
7. .gtf/.gff  [example: Mus_musculus](http://ftp.ensembl.org/pub/release-97/gtf/mus_musculus/Mus_musculus.GRCm38.97.gtf.gz)
